lib/shoulda/matchers/active_model/validate_length_of_matcher.rb

```diff
@@ -368,15 +368,19 @@ def allows_maximum_length?
         end
 
         def allows_length_of?(length, message)
-          allows_value_of(string_of_length(length), message)
+          allows_value_of(string_or_array_of_length(length), message)
         end
 
         def disallows_length_of?(length, message)
-          disallows_value_of(string_of_length(length), message)
+          disallows_value_of(string_or_array_of_length(length), message)
         end
 
-        def string_of_length(length)
-          'x' * length
+        def string_or_array_of_length(length)
+          if @attribute.is_a? String
```

Comment:
Hmm... I'm not sure that this actually works like you think. I believe that `@attribute` here refers to the _name_ of the attribute, not an attribute as a whole.

In order to properly check this, you have to look at the column type. Unfortunately, you can't make the assumption that the model being tested is an ActiveRecord model -- it could be an ActiveModel model, in which case there is no column and hence no type.

There are other matchers in which we do this properly; take a look at https://github.com/thoughtbot/shoulda-matchers/blob/ac9f4822ca25732ccf4d0081d28cac7f19f3b9aa/lib/shoulda/matchers/active_model/validate_inclusion_of_matcher.rb#L517 for inspiration. (Although it shouldn't be that complicated in this case -- you don't need to map `:integer`, for instance, to `:fixnum` or anything like that.)